frontend/apps/reader/modules/readerdevicestatus.lua

```diff
@@ -12,114 +16,282 @@ local ReaderDeviceStatus = InputContainer:new{
+    self.has_battery = powerd:getCapacity() > 0 or powerd:isCharging()
+    if self.has_battery then
+        self.battery_interval = G_reader_settings:readSetting("device_status_battery_interval") or 10
+        self.battery_threshold = G_reader_settings:readSetting("device_status_battery_threshold") or 20
         self.checkLowBattery = function()
-            local threshold = G_reader_settings:readSetting("low_battery_threshold") or 20
             local battery_capacity = powerd:getCapacity()
             if powerd:isCharging() then
-                powerd:setDissmisBatteryStatus(false)
-            elseif powerd:getDissmisBatteryStatus() ~= true and battery_capacity <= threshold then
-                local low_battery_info
-                low_battery_info = ButtonDialogTitle:new {
-                    modal = true,
-                    title = T(_("The battery is getting low.\n%1% remaining."), battery_capacity),
-                    title_align = "center",
-                    title_face = Font:getFace("infofont"),
+                powerd:setDismissBatteryStatus(false)
+            elseif powerd:getDismissBatteryStatus() ~= true and battery_capacity <= self.battery_threshold then
+                powerd:setDismissBatteryStatus(true)
+                UIManager:show(ConfirmBox:new {
+                    text = T(_("Low battery level: %1%\n\nDismiss low battery alarm?"), battery_capacity),
+                    ok_text = _("Dismiss"),
                     dismissable = false,
-                    buttons = {
-                        {
-                            {
-                                text = _("Dismiss"),
-                                callback = function()
-                                    UIManager:close(low_battery_info)
-                                    powerd:setDissmisBatteryStatus(true)
-                                    UIManager:scheduleIn(300, self.checkLowBattery)
-                                end,
-                            },
-                        },
-                    }
-                }
-                UIManager:show(low_battery_info)
-                return
-            elseif powerd:getDissmisBatteryStatus() and battery_capacity > threshold then
-                powerd:setDissmisBatteryStatus(false)
+                    cancel_callback = function()
+                        powerd:setDismissBatteryStatus(false)
+                    end,
+                })
+            elseif powerd:getDismissBatteryStatus() and battery_capacity > self.battery_threshold then
+                powerd:setDismissBatteryStatus(false)
             end
-            UIManager:scheduleIn(300, self.checkLowBattery)
+            UIManager:scheduleIn(self.battery_interval * 60, self.checkLowBattery)
         end
-        self.ui.menu:registerToMainMenu(self)
         self:startBatteryChecker()
-    else
-        self.checkLowBattery = nil
     end
- end
+
+    if not Device:isAndroid() then
```

Comment:
I think @pazos meant something closer to `if Device:hasAbsurdlyLowMemory() then`, although presumably more for enabling G_reader_settings:isTrue("device_status_memory_alarm") by default on a very select few devices than for disabling the functionality completely.

Or is this line about the `io.open("/proc/self/statm", "r")` stuff?

Reply:
Please point me to the criteria of removing (from menu and init) the memory check for some devices.
Both battery and memory checks are disabled (will not start) by default for all devices.

Reply:
> io.open("/proc/self/statm", "r")

This code is taken from the readerfooter module. Are there any restrictions of using it?

Reply:
> Both battery and memory checks are disabled (will not start) by default for all devices.

Right, the question's about the reverse. It's not useful on almost all devices, but apparently it is on yours and possibly some other Kindles.

As an aside, instead of this restarting business, have you tried simply doing this instead when your high memory condition is met?

```
collectgarbage()
collectgarbage()
```

> This code is taken from the readerfooter module. Are there any restrictions of using it?

I'm simply asking what your intent is with line. If it is about the statm thing, then it might make sense to do use the
```
local statm = io.open("/proc/self/statm", "r")
if statm then
```
as the check. Possibly in generic Device or something.

Reply:
> collectgarbage()
> collectgarbage()

Tried at M:55 with absolutely no effect.

Reply:
Yes, collectgarbage() only reclaims some obvious stuff - but we still have some memory leaks somewhere (https://github.com/koreader/koreader/issues/5998#issuecomment-602180998).

For info, on my Kobo, I have my own high memory usage notification. at some hardcoded value of M:300 (and I know I should still be ok up to 380 - but things begin to get slower at 120/150...)

Reply:
> we still have some memory leaks somewhere [1]

Every Document instance ;o). (Possibly every instance of everything, period ^^). c.f., the lru experiment.

Reply:
Hadn't you fixed that ? :)

Reply:
(Haven't yet checked how the latest CoolReader PR's latest commits help - they are supposed to fix some CRE valgrind warnings we've been seeing for a long time.)

Reply:
Nope, it's not engine related, it's (probably) related to our inheritance system and anchored stale pointers *everywhere* ^^.